src/components/data/DataTable.js

```diff
@@ -0,0 +1,184 @@
+  rows = [],
+  title,
+  selectedRow,
+  loading = false,
+  renderItem = (row, field) => row[field],
+  onSelectRow,
+  onConfirmRow,
+  emptyMessage,
+
+  ...htmlAttributes
+}) {
+  const tableRef = useSyncedRef(elementRef);
+  const scrollContext = useContext(ScrollContext);
+
+  useArrowKeyNavigation(tableRef, {
+    selector: 'tbody tr',
+    horizontal: true,
+    vertical: true,
+  });
+
+  const noContent = loading || (!rows.length && emptyMessage);
+  const fields = columns.map(column => column.field);
+
+  /** @param {Row} row */
+  function selectRow(row) {
+    onSelectRow?.(row);
+  }
+
+  /** @param {Row} row */
+  function confirmRow(row) {
+    onConfirmRow?.(row);
+  }
+
+  /**
+   * @param {KeyboardEvent} event
+   * @param {Row} row
+   * */
+  function handleKeyDown(event, row) {
+    if (event.key === 'Enter') {
+      confirmRow(row);
+      event.preventDefault();
+      event.stopPropagation();
+    }
+  }
+
+  // Ensure that a selected row is visible when this table is within
+  // a scrolling context
+  useEffect(() => {
+    if (!selectedRow || !scrollContext) {
+      return;
+    }
+    const scrollEl = scrollContext.scrollRef.current;
+    const tableHead = tableRef.current?.querySelector('thead');
+    /** @type {HTMLElement | undefined | null} */
+    const selectedRowEl = tableRef.current?.querySelector(
+      'tr[aria-selected="true"]'
+    );
+
+    if (scrollEl && selectedRowEl) {
+      // Ensure the row is visible within the scroll content area
+      const scrollOffset = selectedRowEl.offsetTop - scrollEl.scrollTop;
+      if (scrollOffset > scrollEl.clientHeight) {
+        selectedRowEl.scrollIntoView();
+      }
+      // Ensure the row is not obscured by a sticky header
+      if (tableHead) {
+        const headingHeight = tableHead.clientHeight;
+        const headingOffset = scrollOffset - headingHeight;
+        if (headingOffset < 0) {
+          scrollEl.scrollBy(0, headingOffset);
+        }
+      }
+    }
+  }, [selectedRow, tableRef, scrollContext]);
+
+  return (
+    <Table
+      {...htmlAttributes}
+      title={title}
+      elementRef={downcastRef(tableRef)}
+      interactive={!!(onSelectRow || onConfirmRow)}
+      stickyHeader={true}
+      role="grid"
+      data-composite-component="DataTable"
+    >
+      <TableHead>
+        <TableRow>
+          {columns.map(column => (
+            <TableCell key={column.field} classes={column.classes}>
+              {column.label}
+            </TableCell>
+          ))}
+        </TableRow>
+      </TableHead>
+      <TableBody>
+        {!loading &&
+          rows.map((row, idx) => (
+            <TableRow
+              key={idx}
+              selected={row === selectedRow}
+              onClick={() => selectRow(row)}
+              onFocus={() => selectRow(row)}
+              onDblClick={() => confirmRow(row)}
+              onKeyDown={event => handleKeyDown(event, row)}
+            >
+              {fields?.map(field => (
```

Comment:
```suggestion
              {fields.map(field => (
```

`fields` is always an array.